### PR TITLE
fix new dygraph record event

### PR DESCRIPTION
**Pull request by rainyfly**
### PR types

Others
### PR changes
Others
### Describe

1. 将新动态图中对Op内部子过程的打点类型更新为OperatorInner,使得统计表单能正常识别Op。

### CODE REVIEW

**Review by liutiexing — DISMISSED** [review]

File: paddle/fluid/eager/auto_code_generator/eager_generator.cc [hunk 1]
```diff
@@ -2481,7 +2481,7 @@ static std::string GenerateGradNodeHeaderContents(
       "%s\n"
       "    SetIsTensorWrappersCleared(true);\n"
       "  }\n"
-      "  std::string name() override { return \" GradNode%s \"; } \n "
```
Comment (jim19930609) [inline paddle/fluid/eager/auto_code_generator/eager_generator.cc:2484]: 1. "name()" returns the class name, you should change the class name as well.
2. Per coding style, better named as  "GradNode%sMid" or so.

File: paddle/fluid/eager/auto_code_generator/final_state_generator/codegen_utils.py [hunk 2]
```diff
@@ -136,7 +136,7 @@ def RemoveConstAndReference(string):
 
 
 def GetGradNodeName(string):
-    return f"FinalGradNode{string}"
+    return f"{string}_final_grad_node"
```
Comment (jim19930609) [inline paddle/fluid/eager/auto_code_generator/final_state_generator/codegen_utils.py]: Ditto

File: paddle/fluid/eager/backward.cc [hunk 3]
```diff
@@ -643,7 +643,7 @@ std::vector<paddle::experimental::Tensor> RunBackward(
     VLOG(6) << "Running GradNode:" << node->name();
 
     paddle::platform::RecordEvent node_record_event(
-        std::string(typeid(*node).name()) + " grad_node",
```
Comment (jim19930609) [inline paddle/fluid/eager/backward.cc:646]: Some performance analysis script rely on this " grad_node" suffix to identify GradNode runtime. Please keep this suffix unchanged.

File: paddle/fluid/eager/auto_code_generator/eager_generator.cc (hunk 1 above)
Comment (rainyfly) [reply paddle/fluid/eager/auto_code_generator/eager_generator.cc:2484]: Done

File: paddle/fluid/eager/auto_code_generator/final_state_generator/codegen_utils.py (hunk 2 above)
Comment (rainyfly) [reply paddle/fluid/eager/auto_code_generator/final_state_generator/codegen_utils.py]: Done

File: paddle/fluid/eager/backward.cc (hunk 3 above)
Comment (rainyfly) [reply paddle/fluid/eager/backward.cc:646]: Done

**Review by jim19930609 — APPROVED** [review]

**Pull request merged**